pkg/domain/infra/abi/images.go

```diff
@@ -85,6 +84,110 @@ func (ir *ImageEngine) History(ctx context.Context, nameOrID string, opts entiti
 	return &history, nil
 }
 
+func (ir *ImageEngine) Mount(ctx context.Context, nameOrIDs []string, opts entities.ImageMountOptions) ([]*entities.ImageMountReport, error) {
+	var (
+		images []*image.Image
+		err    error
+	)
+	if os.Geteuid() != 0 {
```

Comment:
I think that's dead code at the moment, as we're always uid 0 in the user namespace.

Can we do a rootless check instead?

Reply:
for `mount` we don't automatically join the namespace, so we can raise a proper error when it is used outside of a `podman unshare` session